sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2217,13 +2218,23 @@ object GpuOverrides {
         override def convertToGpu(child: Expression): GpuExpression = GpuUnscaledValue(child)
       }),
     expr[MakeDecimal](
-      "Create a Decimal from an unscaled long value form some aggregation optimizations",
+      "Create a Decimal from an unscaled long value for some aggregation optimizations",
       ExprChecks.unaryProject(TypeSig.DECIMAL, TypeSig.DECIMAL,
         TypeSig.LONG, TypeSig.LONG),
       (a, conf, p, r) => new UnaryExprMeta[MakeDecimal](a, conf, p, r) {
         override def convertToGpu(child: Expression): GpuExpression =
           GpuMakeDecimal(child, a.precision, a.scale, a.nullOnOverflow)
       }),
+    expr[ScalarSubquery](
+      "Subquery that will return only one row and one column",
+      ExprChecks.projectOnly(
+        TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL,
+        TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL,
+        Nil, None),
+      (a, conf, p, r) => new ExprMeta[ScalarSubquery](a, conf, p, r) {
+        override def convertToGpu(): GpuExpression = GpuScalarSubquery(a.plan, a.exprId)
```

Comment:
Sorry I forgot to ask how does `a.plan` get converted to the GPU? Do your test show that it ends up on the GPU?

Reply:
`a.plan` here is `BaseSubqueryExec(child: subqueryExec)`. 
I think the child plan will go through GpuOverriding rules as an [independent physical plan](https://github.com/apache/spark/blob/branch-3.1/sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala#L178) before contained with `BaseSubqueryExec`.
And the child is not necessarily on the GPU if I am not wrong. I think the avdantage of `GpuScalarSubquery` is making the outside plan (usually FilterExec) who owns scalar_subquery expression GPU-enabled if there is no other fallback reason.